src/AaveCurator/AaveCurator.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {console2} from 'forge-std/Test.sol';
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {Ownable} from 'solidity-utils/contracts/oz-common/Ownable.sol';
+import {Initializable} from 'solidity-utils/contracts/transparent-proxy/Initializable.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {AaveV2Ethereum, AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveGovernanceV2} from 'aave-address-book/AaveGovernanceV2.sol';
+
+import {IWeth} from './interfaces/IWeth.sol';
+import {IWstEth} from './interfaces/IWstEth.sol';
+import {ILido} from './interfaces/ILido.sol';
+import {IMilkman} from './interfaces/IMilkman.sol';
+import {IRocketPoolDeposit} from './interfaces/IRocketPoolDeposit.sol';
+
+contract AaveCurator is Initializable, Ownable {
+  using SafeERC20 for IERC20;
+
+  event DepositedIntoV2(address indexed token, uint256 amount);
+  event DepositedIntoV3(address indexed token, uint256 amount);
+  event ManagerChanged(address indexed oldManager, address indexed newManager);
+  event SwapCanceled(address fromToken, address toToken, uint256 amount);
+  event SwapRequested(address fromToken, address toToken, uint256 amount);
+  event TokenUpdated(address indexed token, bool allowed);
+
+  error Invalid0xAddress();
+  error InvalidAmount();
+  error InvalidCaller();
+  error InvalidOracleType();
+  error InvalidRecipient();
+  error InvalidToken();
+  error OracleNotSet();
+
+  enum TokenType {
+    Standard,
+    BPT8020
+  }
+
+  struct TokenChailinkOracleData {
+    address oracle;
+    bool isEthBased;
+  }
+
+  modifier onlyOwnerOrManager() {
+    if (msg.sender != owner() && msg.sender != manager) {
+      revert InvalidCaller();
+    }
+    _;
+  }
+
+  IRocketPoolDeposit private constant ROCKET_POOL =
+    IRocketPoolDeposit(0xDD3f50F8A6CafbE9b31a427582963f465E745AF8);
+  IWeth private constant WETH = IWeth(AaveV2EthereumAssets.WETH_UNDERLYING);
+
+  address public constant STETH = AaveV2EthereumAssets.stETH_UNDERLYING;
+  address public constant BPT_PRICE_CHECKER = 0x7961bBC81352F26d073aA795EED51290C350D404; // TODO: Update with new one
+  address public constant CHAINLINK_PRICE_CHECKER = 0x4D2c3773E69cB69963bFd376e538eC754409ACFa;
+
+  address public milkman = 0x11C76AD590ABDFFCD980afEC9ad951B160F02797;
+  address public manager;
+
+  mapping(address tokenAddress => bool) public allowedFromTokens;
```

Comment:
Yeah, it's just to avoid the guardian/manager from interacting with bad tokens.
The allowedTO tokens is so that the manager cannot swap for tokens the DAO doesn't approve of first.